clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java

```diff
@@ -54,88 +59,188 @@ private static Task getTask() {
 
   @Override
   public DeploymentResult operate(List priorOutputs) {
-    getTask().updateStatus(BASE_PHASE, "Initializing Create Amazon ECS Server Group Operation...");
-    AmazonCredentials credentials = (AmazonCredentials) accountCredentialsProvider.getCredentials(description.getCredentialAccount());
-    String versionString = inferServergroupVersion();
+    updateTaskStatus("Initializing Create Amazon ECS Server Group Operation...");
```

Comment:
I'd like to have all `updateTaskStatus` calls on the same abstraction level, so inside the `operate` method, like line 62

Reply:
Having all the tasks in the operate method is suboptimal. Some of those task statements require to have knowledge of the details of the operation that they are covering. i.e.
```
updateTaskStatus(String.format("Creating %s of %s with %s for %s.",
      desiredCapacity, serviceName, taskDefinitionArn, description.getCredentialAccount()));
```

Also, later on grep'ing for those strings would require an extra jump to find out the logic that logged it.